lib/analytics.js

```diff
@@ -708,25 +709,27 @@ Analytics.prototype._invoke = function(method, facade) {
     } else {
       try {
         // Apply any integration middlewares that exist, then invoke the integration with the result.
-        self._integrationMiddlewares.applyMiddlewares(facadeCopy.obj, function(
-          payload
-        ) {
-          // A nullified payload should not be sent to an integration.
-          if (payload === null) {
-            self.log('Payload to integration "%s" was null and dropped.', name);
-            return;
+        self._integrationMiddlewares.applyMiddlewares(
+          facadeCopy,
+          integration.name,
+          function(facadeResult) {
+            // A nullified payload should not be sent to an integration.
+            if (facadeResult === null) {
```

Comment:
Is this guaranteed to be of the facade type?

Reply:
Ahh, good catch!